### PR TITLE
Enable more voice options

### PR DESCRIPTION
**Pull request by steinbro**
I found another registry key to search that enabled the newer Mark voice, as well as mobile versions of David and Zira.

Digging through the registry, I also found Eva, the Cortana voice, but only in a key that looked to be machine-specific, so I didn't include it. There are hacks to get this voice generally available as a system TTS voice that involve copying the value to the keys we currently search: https://github.com/watermarkhu/enable-microsoft-eva-tts

### CODE REVIEW

File: src/sapi.rs [hunk 1]
```diff
@@ -294,7 +302,9 @@ impl SpVoice {
     }
 
     fn set_voice(&mut self, token: Speech::ISpObjectToken) {
-        unsafe { self.voice.SetVoice(&token).expect("set voice") }
+        unsafe {
+            self.voice.SetVoice(&token).ok();
```
Comment (Eh2406) [inline src/sapi.rs]: Why is this change necessary?

File: src/sapi.rs [hunk 2]
```diff
@@ -259,27 +259,35 @@ impl SpVoice {
...
+            for c in voice_categories.iter() {
+                let category: Speech::ISpObjectTokenCategory = syscom::CoCreateInstance(
+                    &Speech::SpObjectTokenCategory,
+                    None,
+                    syscom::CLSCTX_ALL,
                 )
-                .expect("set voice category id");
-
-            let token_enum = category.EnumTokens(w!(""), w!("")).expect("get voice list");
-            loop {
```
Comment (Eh2406) [inline src/sapi.rs:274]: I wonder if this loop could be cleaned up with [std::iter::from_fn](https://doc.rust-lang.org/std/iter/fn.from_fn.html), what do you think?

**Review by Eh2406 — APPROVED** [review]
This is an excellent find! I'm excited to try the newer voices!

File: src/sapi.rs (hunk 1 above)
Comment (steinbro) [reply src/sapi.rs]: It's not strictly required -- I changed it when I found some voices in the registry that crashed rust-reader when selected. This line wasn't the culprit, but I figured that failure to change the voice should not be a fatal error. Do you agree?

File: src/sapi.rs (hunk 2 above)
Comment (steinbro) [reply src/sapi.rs:274]: Huh, that's a good function to know about. This does seem like the appropriate scenario. Is my new change what you had in mind?

File: src/sapi.rs (hunk 1 above)
Comment (Eh2406) [reply src/sapi.rs]: Seems like a reasonable change! Although we should probably be consistent about whether `set_` error are fatal. An alternative is for all of the `set_` to return a Result, and let the caller figure out what to do with the error. (As I said, error handling in this code base isn't great.)

File: src/sapi.rs (hunk 2 above)
Comment (Eh2406) [reply src/sapi.rs:274]: Yes, that's exactly what I had in mind.

File: src/sapi.rs (hunk 1 above)
Comment (Eh2406) [reply src/sapi.rs]: Edit: this looks good, sorry. It's only used a couple of lines further down, and we handle the error by returning the name of the voice that is currently set.
